Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -124,51 +94,111 @@ extension HTTPClient {
+    }
+
+    /// Splits this string at the first "=" sign, and trims whitespace from each side.
+    fileprivate func parseKeyValuePair() -> (key: SubSequence, value: SubSequence)? {
+        guard let keyValueSeparator = self.firstIndex(of: "=") else {
+            return nil
+        }
+        let trimmedName = self[..<keyValueSeparator].trimmingWhitespace()
+        let trimmedValue = self[self.index(after: keyValueSeparator)...].trimmingWhitespace()
+        return (trimmedName, trimmedValue)
+    }
+
+    /// If this string is a key-value pair, splits it and normalizes the key.
+    /// Otherwise, return this string, trimmed of whitespace and normalized to lowercase.
+    fileprivate func parseCookieComponent() -> (String?, String?) {
+        guard let (trimmedName, trimmedValue) = self.parseKeyValuePair() else {
+            let trimmedName = self.trimmingWhitespace()
+            return (trimmedName.isEmpty ? nil : trimmedName.lowercased(), nil)
+        }
+        guard !trimmedName.isEmpty else {
+            return (nil, nil)
+        }
+        return (trimmedName.lowercased(), trimmedValue.isEmpty ? nil : String(trimmedValue))
     }
 }
 
-extension HTTPClient.Response {
-    /// List of HTTP cookies returned by the server.
-    public var cookies: [HTTPClient.Cookie] {
-        return self.headers["set-cookie"].compactMap { HTTPClient.Cookie(header: $0, defaultDomain: self.host) }
+#if canImport(Darwin)
+import Darwin
+#elseif canImport(Glibc)
+import Glibc
+#endif
+import CShims
+
+private let posixLocale: locale_t = {
+    // All POSIX systems must provide a "POSIX" locale, and its date/time formats are US English.
+    // https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap07.html#tag_07_03_05
+    newlocale(LC_TIME_MASK | LC_NUMERIC_MASK, "POSIX", nil)!
+}()
+
+private func parseCookieTime(_ string: String) -> Int64? {
+    var timeComps = tm()
+    var hasExplicitTimeZone = false
+    parse: do {
+        if swiftahc_cshims_strptime_l(string, "%a, %d %b %Y %H:%M:%S %Z", &timeComps, .init(posixLocale)) != nil {
+            hasExplicitTimeZone = true
+            break parse
+        }
+        timeComps = tm()
+        if swiftahc_cshims_strptime_l(string, "%a, %d-%b-%y %H:%M:%S %Z", &timeComps, .init(posixLocale)) != nil {
+            hasExplicitTimeZone = true
+            break parse
+        }
+        timeComps = tm()
+        if swiftahc_cshims_strptime_l(string, "%a %b %d %H:%M:%S %Y", &timeComps, .init(posixLocale)) != nil {
+            break parse
+        }
+        return nil
+    }
+    #if canImport(Darwin)
+    // Darwin converts values with explicit time-zones to the local time-zone. Glibc appears not to.
+    // https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/strptime.3.html
+    if hasExplicitTimeZone {
+        let timestamp = Int64(mktime(&timeComps))
+        return timestamp == -1 && errno == EOVERFLOW ? nil : timestamp
```

Comment:
We shouldn't have to do this at all: cookie expires are supposed to always be GMT, never local time. Is there a reason we think we need it instead of just only accepting GMT?

Reply:
In practice, most implementations of `strptime` seem to only recognise GMT and sometimes also the current timezone (including Darwin). "%Z" itself isn't actually part of the POSIX spec which is why Darwin and Glibc have different behaviour with regards to converting to local time.

If we only want to support GMT, we could always just look for the literal string "GMT" and chop it off, avoiding all of this. Actually that sounds obviously worthwhile, now that I spell it out like this.

Reply:
Literal string it is!